.github/workflows/ci-pr.yml

```diff
@@ -279,6 +298,28 @@ jobs:
               TestAlgohWithExpect \
               TestGoalWithExpect \
               TestTealdbgWithExpect
+      - name: Determine partition total
+        if: ${{ !cancelled() && matrix.test_type != 'e2e_subs' }}
+        shell: bash
+        run: |
+          case "${{ matrix.test_type }}" in
+            test)
+              echo "PARTITION_TOTAL=6" >> $GITHUB_ENV
+              ;;
+            integration)
+              echo "PARTITION_TOTAL=5" >> $GITHUB_ENV
+              ;;
+            e2e_expect)
+              echo "PARTITION_TOTAL=10" >> $GITHUB_ENV
```

Comment:
And then these need to match above?  Can the simple numbers be variables of some sort? (I doubt anything is possible with the lists)